pwnlib/util/packing.py

```diff
@@ -481,34 +484,41 @@ def make_unpacker(word_size = None, endianness = None, sign = None, **kwargs):
 
 
 
-def _flat(args, preprocessor, packer):
-    out = []
+def _flat(args, preprocessor, packer, address):
+    out = io.BytesIO()
     for arg in args:
+        address = address + out.tell()
```

Comment:
This looks wrong to me: Isn't `out.tell()` the amount of data written since the beginning, but `address` here is the address of the last written argument? I think it should be `initial_address + out.tell()` where `initial_address` is set to the value of the `address` parameter at the start of the function.

Reply:
I think this explains the wrong output of the second sample.